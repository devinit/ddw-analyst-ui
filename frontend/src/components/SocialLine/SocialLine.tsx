import * as React from 'react';
import { Button } from 'react-bootstrap';
import { ConditionalRender } from '../ConditionalRender';
import { FunctionComponent } from 'react';

interface SocialLineProps {
  google?: boolean;
  facebook?: boolean;
  twitter?: boolean;
  onGoogle?: () => void;
  onFacebook?: () => void;
  onTwitter?: () => void;
}

export const SocialLine: FunctionComponent<SocialLineProps> = (props) => {
  return (
    <div className="social-line">
      <ConditionalRender render={!!props.google}>
        <Button variant="link" className="btn-white" onClick={props.onGoogle}>
          <i className="fa fa-google" />
        </Button>
      </ConditionalRender>
      <ConditionalRender render={!!props.facebook}>
        <Button variant="link" className="btn-white" onClick={props.onFacebook}>
          <i className="fa fa-facebook" />
        </Button>
      </ConditionalRender>
      <ConditionalRender render={!!props.twitter}>
        <Button variant="link" className="btn-white" onClick={props.onTwitter}>
          <i className="fa fa-twitter" />
        </Button>
      </ConditionalRender>
    </div>
  );
};
