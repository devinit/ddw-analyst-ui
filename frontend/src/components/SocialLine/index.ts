import { SocialLine } from './SocialLine';

export default SocialLine;
export * from './SocialLine';
