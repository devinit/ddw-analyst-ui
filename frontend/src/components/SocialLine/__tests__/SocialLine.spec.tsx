import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SocialLine } from '../SocialLine';

test('renders only the wrapper when no social options are specified', () => {
  const renderer = TestRenderer.create(<SocialLine />).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders the google button when google is set to true', () => {
  const renderer = TestRenderer.create(<SocialLine google />).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders the facebook button when facebook is set to true', () => {
  const renderer = TestRenderer.create(<SocialLine facebook />).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders the twitter button when twitter is set to true', () => {
  const renderer = TestRenderer.create(<SocialLine twitter />).toJSON();

  expect(renderer).toMatchSnapshot();
});
