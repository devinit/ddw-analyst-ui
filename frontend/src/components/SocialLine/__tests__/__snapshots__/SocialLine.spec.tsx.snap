// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders only the wrapper when no social options are specified 1`] = `
<div
  className="social-line"
/>
`;

exports[`renders the facebook button when facebook is set to true 1`] = `
<div
  className="social-line"
>
  <button
    className="btn-white btn btn-link"
    disabled={false}
    type="button"
  >
    <i
      className="fa fa-facebook"
    />
  </button>
</div>
`;

exports[`renders the google button when google is set to true 1`] = `
<div
  className="social-line"
>
  <button
    className="btn-white btn btn-link"
    disabled={false}
    type="button"
  >
    <i
      className="fa fa-google"
    />
  </button>
</div>
`;

exports[`renders the twitter button when twitter is set to true 1`] = `
<div
  className="social-line"
>
  <button
    className="btn-white btn btn-link"
    disabled={false}
    type="button"
  >
    <i
      className="fa fa-twitter"
    />
  </button>
</div>
`;
